api/applications/views.py

```diff
@@ -98,3 +105,31 @@ def perform_update(self, serializer):
         serializer.validated_data['owner'] = self.request.user
         # TODO: Write code to transfer ownership
         serializer.save(owner=self.request.user)
+
+
+class ApplicationReset(generics.RetrieveUpdateDestroyAPIView, ApplicationMixin):
```

Comment:
Why is there a separate view for `ApplicationReset`? Why not do this as part of the update method in the `ApplicationDetail` view?